protocol/communities/manager.go

```diff
@@ -957,6 +1045,490 @@ func (m *Manager) GetAdminCommunitiesChatIDs() (map[string]bool, error) {
+			break
+		}
+		log.Println("Creating message archive for partition (", partition, "). From: ", from, " To: ", to)
+		messages, err := m.persistence.GetWakuMessagesByFilterTopic(topics, uint64(from.Unix()), uint64(to.Unix()))
+		if err != nil {
+			return err
+		}
+
+		wakuMessageArchive := m.createWakuMessageArchive(from, to, messages, topicsAsByteArrays)
+		encodedArchive, err := proto.Marshal(wakuMessageArchive)
+		if err != nil {
+			return err
+		}
+
+		rawSize := len(encodedArchive)
+		padding := 0
+		size := 0
+
+		if rawSize > pieceLength {
+			size = rawSize + pieceLength - (rawSize % pieceLength)
+			padding = size - rawSize
+		} else {
+			padding = pieceLength - rawSize
+			size = rawSize + padding
+		}
+
+		wakuMessageArchiveIndexMetadata := &protobuf.WakuMessageArchiveIndexMetadata{
+			Metadata: wakuMessageArchive.Metadata,
+			Offset:   offset,
+			Size:     uint64(size),
+			Padding:  uint64(padding),
+		}
+
+		wakuMessageArchiveIndexMetadataBytes, err := proto.Marshal(wakuMessageArchiveIndexMetadata)
+		if err != nil {
+			return err
+		}
+
+		wakuMessageArchiveIndex[crypto.Keccak256Hash(wakuMessageArchiveIndexMetadataBytes).String()] = wakuMessageArchiveIndexMetadata
+		encodedArchives = append(encodedArchives, &EncodedArchiveData{bytes: encodedArchive, padding: padding})
+		from = to
+		to = to.Add(partition)
+		if to.After(endDate) {
+			to = endDate
+		}
+		offset = offset + uint64(rawSize) + uint64(padding)
+	}
+
+	if len(encodedArchives) == 0 {
+		log.Println("No new archives created")
+		return nil
+	}
+
+	dataBytes := make([]byte, 0)
+	if _, err := os.Stat(dataPath); err == nil {
+		dataBytes, err = os.ReadFile(dataPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, encodedArchiveData := range encodedArchives {
+		dataBytes = append(dataBytes, encodedArchiveData.bytes...)
+		dataBytes = append(dataBytes, make([]byte, encodedArchiveData.padding)...)
+	}
+
+	wakuMessageArchiveIndexProto.Archives = wakuMessageArchiveIndex
+	indexBytes, err := proto.Marshal(wakuMessageArchiveIndexProto)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(indexPath, indexBytes, 0644)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(dataPath, dataBytes, 0644)
+	if err != nil {
+		return err
+	}
+
+	metaInfo := metainfo.MetaInfo{
+		AnnounceList: defaultAnnounceList,
+	}
+	metaInfo.SetDefaults()
+	metaInfo.CreatedBy = common.PubkeyToHex(m.identity)
+
+	info := metainfo.Info{
+		PieceLength: int64(pieceLength),
+	}
+
+	err = info.BuildFromFilePath(archiveDir)
+	if err != nil {
+		return err
+	}
+
+	metaInfo.InfoBytes, err = bencode.Marshal(info)
+	if err != nil {
+		return err
+	}
+
+	metaInfoBytes, err := bencode.Marshal(metaInfo)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(m.torrentFile(communityID.String()), metaInfoBytes, 0644)
+	if err != nil {
+		return err
+	}
+
+	err = m.persistence.UpdateLastMessageArchiveEndDate(communityID, uint64(from.Unix()))
```

Comment:
This has a bug. We need to check whether we already have one to update it. Otherwise add it.